### ISSUE TITLE
feat(button): Consider enter-key by default

### ISSUE THREAD

**Issue by andreElrico** [angular/components#24503]
### Feature Description

with a native html button enter/return key will "click/ interact" wtih the button.

why is that not considered with the material buttons. From an a11y standpoint it makes sense to have this feature.

Current workaround:

`<button mat-button (click)="onClick()" (keyup.enter)="onClick()">Basic</button>
`

if https://github.com/angular/angular/issues/4059 is solved, material button could implement that on (click). Otherwise I suggest to have an `(interacted)` for click/touch and keyboard interaction.

### Use Case

Basically (click) would mirror the native behaviour.

**Comment by crisbeto** [comment]
This should be the case already. We don't do anything in `mat-button` to prevent the default behavior. Maybe you're setting the `mat-button` attribute on a node that isn't a `button`?

**Comment by andreElrico** [comment]
I have tested it on a stackblitz again. It works as expected. Closing this for now. There must be a problem with my usecase. Going to investigate ...
